File: v2/src/main/java/com/api/v2/doctors/domain/DoctorAuditTrail.java
package com.api.v2.doctors.domain;

import com.api.v2.doctors.domain.exposed.Doctor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.UUID;

@Document
public record DoctorAuditTrail(
        @Id
        String id,
        Doctor doctor,
        LocalDateTime createdAAt
) {

    public static DoctorAuditTrail of(Doctor doctor) {
        return new DoctorAuditTrail(
                UUID.randomUUID().toString(),
                doctor,
                LocalDateTime.now()
        );
    }

}

File: v2/src/main/kotlin/com/api/v2/doctors/domain/DoctorAuditTrail.kt
package com.api.v2.doctors.domain

import com.api.v2.doctors.Doctor
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.time.LocalDateTime
import java.util.UUID

@Document(collection = "DoctorAuditTrail")
data class DoctorAuditTrail(
    @Id
    val id: String,
    val doctor: Doctor,
    val createdAt: LocalDateTime
) {

    companion object {
        fun of(doctor: Doctor): DoctorAuditTrail {
            return DoctorAuditTrail(
                UUID.randomUUID().toString(),
                doctor,
                LocalDateTime.now()
            )
        }
    }
}
